src/lib/components/Button.svelte

```diff
@@ -0,0 +1,66 @@
+<script lang="ts">
+import type { Snippet } from "svelte";
+import Loader from "./Loader.svelte";
+
+type ButtonProps = {
+    variant?: "default" | "secondary" | "destructive" | "outline" | "ghost" | "link";
+    size?: "default" | "icon" | "sm" | "lg";
```

Comment:
[sug] The names "default" and "icon" under size don’t give me a clear idea of whether they’re larger or smaller than sm or lg. I think it would be easier to understand if these options followed a consistent size scale